File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Audio/MusicMaster.ts
import { MediaManager } from "csharp"
import {Mathf} from "Core/Module/Math/Mathf";
import FileHelper from "mods/ModTheSpire/Scripts/FileHelper";

export class MusicMaster{
    public static Map:Map<string,string> = new Map<string, string>([
        ["SHOP_BGM", FileHelper.FormatPath("Audio/music/STS_Merchant_NewMix_v1.ogg")],
        ["SHRINE_BGM", FileHelper.FormatPath("Audio/music/STS_Merchant_NewMix_v1.ogg")],
        ["MINDBLOOM_BGM", FileHelper.FormatPath("Audio/music/STS_Shrine_NewMix_v1.ogg")],
        ["LEVEL_1_BOSS_BGM", FileHelper.FormatPath("Audio/music/STS_Boss1MindBloom_v1.ogg")],
        ["LEVEL_2_BOSS_BGM", FileHelper.FormatPath("Audio/music/STS_Boss2MindBloom_v1.ogg")],
        ["LEVEL_3_BOSS_BGM", FileHelper.FormatPath("Audio/music/STS_Boss3MindBloom_v1.ogg")],
        ["LEVEL_4_BOSS_BGM", FileHelper.FormatPath("Audio/music/STS_Boss4_v6.ogg")],
        ["ELITE_BGM", FileHelper.FormatPath("Audio/music/STS_EliteBoss_NewMix_v1.ogg")],
        ["CREDITS", FileHelper.FormatPath("Audio/music/STS_Credits_v5.ogg")],
    ]);

    public static Play(key:string){
        let bgm = this.Map.get(key);
        if(bgm){
            MediaManager.PlayMusic(bgm,true);
        }
    }
    
    public static Silence(){
        
    }
    
    public static FadeOut(){
        
    }
}

File: Unity/Assets/Scripts/ModTheSpire/MusicMaster.cs
namespace ModTheSpire
{
    public class MusicMaster
    {
        public static MusicMaster SHOP_BGM= new MusicMaster("Audio/music/STS_Merchant_NewMix_v1.ogg");
        public static MusicMaster SHRINE_BGM= new MusicMaster("Audio/music/STS_Merchant_NewMix_v1.ogg");
        public static MusicMaster MINDBLOOM_BGM= new MusicMaster("Audio/music/STS_Shrine_NewMix_v1.ogg");
        public static MusicMaster LEVEL_1_BOSS_BGM= new MusicMaster("Audio/music/STS_Boss1MindBloom_v1.ogg");
        public static MusicMaster LEVEL_2_BOSS_BGM= new MusicMaster("Audio/music/STS_Boss2MindBloom_v1.ogg");
        public static MusicMaster LEVEL_3_BOSS_BGM= new MusicMaster("Audio/music/STS_Boss3MindBloom_v1.ogg");
        public static MusicMaster LEVEL_4_BOSS_BGM= new MusicMaster("Audio/music/STS_Boss4_v6.ogg");
        public static MusicMaster ELITE_BGM= new MusicMaster("Audio/music/STS_EliteBoss_NewMix_v1.ogg");
        public static MusicMaster CREDITS= new MusicMaster("Audio/music/STS_Credits_v5.ogg");
            
        public string Key { get; }

        public MusicMaster(string key)
        {
            Key = key;
        }


        public static void Play(MusicMaster key)
        {
            MediaManager.PlayMusic(key.Key, true);
        }
    
        public static void Silence(){
        
        }
    
        public static void FadeOut(){
        
        }
    }
}